custom_components/kia_uvo/KiaUvoApiCA.py

```diff
@@ -123,25 +123,31 @@ def get_cached_vehicle_status(self, token: Token):
 
         vehicle_status["odometer"] = {}
         vehicle_status["odometer"]["unit"]= response["currentOdometerUnit"]
-        vehicle_status["odometer"]["value"]= response["currentOdometer"]
+        vehicle_status["odometer"]["value"]= response["currentOdometer"]       
         vehicle_status["vehicleLocation"] = self.get_location(token)
-
-            
+             
         return vehicle_status
 
     def get_location(self, token: Token):
         url = self.API_URL + "fndmcr"
         headers = self.API_HEADERS
         headers["accessToken"] = token.access_token
         headers["vehicleId"] = token.vehicle_id
-        headers["pAuth"] = self.get_pin_token(token)
-
-        response = requests.post(url, headers=headers, data=json.dumps({"pin": self.pin}))
-        response = response.json()
-        
-        _LOGGER.debug(f"{DOMAIN} - Get Vehicle Location {response}")
-
-        return response["result"]
+        try:
+            headers["pAuth"] = self.get_pin_token(token)
+
+            response = requests.post(url, headers=headers, data=json.dumps({"pin": self.pin}))
+            response = response.json()
+            _LOGGER.debug(f"{DOMAIN} - Get Vehicle Location {response}")
+            if response["result"]["coord"]["lat"] == None:
```

Comment:
if the key does not exists, it will try another error. key means `lat` or `coord`. i suggest you to check result status and status code. there is always S (succees) and F (failure) exists in the response. until we find a better solution for this, we can disable this call from parent function (you can keep it locally for testing purposes)

if you need a hand, happy to jump in but I do not have any account in CA :) if you can share your car for couple of hours, we can work together. or else, you can dump response of API.

Reply:
I can check but I don't think you can share cars in Canada.  

Are you thinking like this? 

`            if response["responseHeader"]["responseCode"] != 0:
                raise Exception('No Location Located')`

Reply:
Yes similar to this.

Reply:
Done.  This has come along when with the latest commit.   Had to handle a few spots to try make this not break other parts.  So far been pretty stable for me.